lib/screens/v2/widget_instance/dup_alert.ex

```diff
@@ -0,0 +1,299 @@
+    suspension      boundary_upstream   1             2             partial     full_screen partial
+    suspension      boundary_downstream 1             2             partial     full_screen partial
+  # Delay alerts are handled separately, see `defp do_alert_layout({:delay, ...` below.
+  """
+
+  parameters_to_layout =
+    alert_layout_table
+    |> String.split("\n", trim: true)
+    |> Enum.reject(&String.starts_with?(&1, "#"))
+    |> Enum.flat_map(fn line ->
+      [
+        effect,
+        location,
+        affected_line_count,
+        total_line_count,
+        layout_zero,
+        layout_one,
+        layout_two
+      ] = String.split(line, ~r/\s+/, trim: true)
+
+      # Convert strings to appropriate types
+      [effect, location, layout_zero, layout_one, layout_two] =
+        Enum.map(
+          [effect, location, layout_zero, layout_one, layout_two],
+          &String.to_existing_atom/1
+        )
+
+      [affected_line_count, total_line_count] =
+        Enum.map([affected_line_count, total_line_count], &String.to_integer/1)
+
+      base_parameters = {effect, location, affected_line_count, total_line_count}
+
+      [
+        {Tuple.append(base_parameters, :zero), layout_zero},
+        {Tuple.append(base_parameters, :one), layout_one},
+        {Tuple.append(base_parameters, :two), layout_two}
+      ]
+    end)
+    |> Enum.into(%{})
+
+  @type layout_parameters ::
+          {Alert.effect(), location :: atom, affected_line_count :: 1..2,
+           total_line_count :: 1..2, rotation_index}
+
+  # Now, define function clauses for each row in the table.
+  @spec do_alert_layout(layout_parameters) :: :full_screen | :partial | :no_render
+  for {parameters, layout} <- parameters_to_layout do
+    parameters_ast = Macro.escape(parameters)
+    defp do_alert_layout(unquote(parameters_ast)), do: unquote(layout)
+  end
+
+  # Delays always use partial alerts on all 3 rotations of the screen.
+  defp do_alert_layout(
+         {:delay, _location, _affected_line_count, _total_line_count, _rotation_index}
+       ) do
+    :partial
+  end
+
+  # If matching inputs aren't found in the table, we assume the alert isn't relevant and do not render it.
+  defp do_alert_layout(_parameters), do: :no_render
+
+  def get_affected_lines(t) do
+    t
+    |> BaseAlert.informed_routes_at_home_stop()
+    |> routes_to_lines()
+  end
+
+  def get_total_lines(t) do
+    t
+    |> BaseAlert.active_routes_at_stop()
+    |> routes_to_lines()
+  end
+
+  # For certain parts of the DUP alert logic, we're only interested in subway lines,
+  # not routes. This merges all of the "Green-B/C/D/E" routes into just "Green".
+  defp routes_to_lines(route_ids) do
+    route_ids
+    |> Enum.map(fn
+      "Green-" <> _branch -> "Green"
+      other_subway_route -> other_subway_route
+    end)
+    |> Enum.uniq()
+  end
+
+  def alert_ids(%__MODULE__{} = t), do: t.alert.id
```

Comment:
Gives an `Screens.ScreensByAlert.SelfRefreshRunner` error. Should be:

```suggestion
  def alert_ids(%__MODULE__{} = t), do: [t.alert.id]
```